### PR TITLE
Add support for register custom ops

### PR DESCRIPTION
**Pull request by zerollzeng**
Signed-off-by: Zero Zeng <zerollzeng@gmail.com>

### CODE REVIEW

**Comment by zerollzeng** [comment]
Hi guys, I create this PR to add support for c++ registered ops. anyone can help review? @guschmue @TomWildenhain-Microsoft

For the registered ops, we can refer to https://www.tensorflow.org/guide/create_op

I also create a simple sample to demonstrate this feature.

[samples.zip](https://github.com/onnx/tensorflow-onnx/files/6470167/samples.zip)

#### Step to reproduce

1. unzip and cd to samples.
2. cd register_tf_op
3. sh build.sh to build the custom op.
4. cd .. and python custom_op.py to generate the sample saved model
5. python -m tf2onnx.convert --saved-model saved_model/ --output model.onnx
6. you should observe the error cast by tensorflow
```
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.8/dist-packages/tf2onnx-1.9.0-py3.8.egg/tf2onnx/convert.py", line 474, in <module>
    main()
  File "/usr/local/lib/python3.8/dist-packages/tf2onnx-1.9.0-py3.8.egg/tf2onnx/convert.py", line 218, in main
    graph_def, inputs, outputs, initialized_tables, tensors_to_rename = tf_loader.from_saved_model(
  File "/usr/local/lib/python3.8/dist-packages/tf2onnx-1.9.0-py3.8.egg/tf2onnx/tf_loader.py", line 479, in from_saved_model
    _from_saved_model_v2(model_path, input_names, output_names,
  File "/usr/local/lib/python3.8/dist-packages/tf2onnx-1.9.0-py3.8.egg/tf2onnx/tf_loader.py", line 386, in _from_saved_model_v2
    imported = tf.saved_model.load(model_path, tags=tag)  # pylint: disable=no-value-for-parameter
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/saved_model/load.py", line 859, in load
    return load_internal(export_dir, tags, options)["root"]
  File "/usr/local/lib/python3.8/dist-packages/tensorflow/python/saved_model/load.py", line 892, in load_internal
    raise FileNotFoundError(
FileNotFoundError: Op type not registered 'LinearSumAssignmentOp' in binary running on eab4b75bd375. Make sure the Op and Kernel are registered in the binary running in this process. Note that if you are loading a saved graph which used ops from tf.contrib, accessing (e.g.) `tf.contrib.resampler` should be done before importing the graph, as contrib ops are lazily registered when the module is first accessed.
```

7. install the patch I create and run with python -m tf2onnx.convert --saved-model saved_model/ --output model.onnx --registered_ops ./register_tf_op/LinearSumAssignmentOp.so
8. now we can generate the model.onnx contains the custom op.

**Review by TomWildenhain-Microsoft — COMMENTED** [review]
Thanks for the contribution!  I'm fine with this change.  I added a more complete description.  @guschmue are you OK with adding this flag?

File: tf2onnx/convert.py [hunk 1]
```diff
@@ -75,6 +75,7 @@ def get_args():
     parser.add_argument("--custom-ops", help="comma-separated map of custom ops to domains in format OpName:domain")
     parser.add_argument("--extra_opset", default=None,
                         help="extra opset with format like domain:version, e.g. com.microsoft:1")
+    parser.add_argument("--registered_ops", help="comma-separated of registerd ops path")
```
[the remark this answers is not in the document]
Comment (guschmue) [reply tf2onnx/convert.py:78]: I wonder if registered_ops is the best name.
Would 
```
--load_libraries
```
describe it better ?

[the remark this answers is not in the document]
Comment (zerollzeng) [reply tf2onnx/convert.py:78]: what about --load_op_libraries?

[the remark this answers is not in the document]
Comment (guschmue) [reply tf2onnx/convert.py:78]: yes, you got my vote!

**Comment by guschmue** [comment]
@zerollzeng, there is this DCO check that fails. 
Do the next ```git commit``` with a ```-s``` and it should clear.

**Comment by zerollzeng** [comment]
Hi @guschmue, can you help review the update? Thanks.

**Comment by guschmue** [comment]
sure. pylint is complaining about a long line in convert.py. Can you fix it? After that we should be all good.

```
+ pylint --rcfile=tools/pylintrc --ignore=version.py,tflite --disable=cyclic-import tf2onnx tests/backend_test_base.py tests/common.py tests/conftest.py tests/huggingface.py tests/run_pretrained_models.py tests/test_api.py tests/test_backend.py tests/test_cond.py tests/test_const_fold.py tests/test_convert.py tests/test_cudnn_compatible_gru.py tests/test_cudnn.py tests/test_custom_rnncell.py tests/test_example.py tests/test_grublock.py tests/test_gru.py tests/test_internals.py tests/test_loops.py tests/test_lstmblock.py tests/test_lstm.py tests/test_onnx_shape_inference.py tests/test_optimizers.py tests/test_profile.py tests/test_seq2seq.py tests/test_stacked_lstm.py tests/test_string_ops.py tests/test_tflite_postprocess.py tests/test_tflite_utils.py tests/test_tf_shape_inference.py tools -j 0
************* Module tf2onnx.convert
tf2onnx/convert.py:74:0: C0301: Line too long (131/120) (line-too-long)
```

**Comment by zerollzeng** [comment]
Fixed

**Comment by guschmue** [comment]
ah, I just noticed one more problem. Your commits come from the user 'Zero Zeng' which is not a github user. Because of this the cla check is unhappy.
I have not tried this myself but belief the instructions here should work:
https://stackoverflow.com/questions/41349644/commit-with-wrong-username-to-github

**Comment by zerollzeng** [comment]
Thanks for reminding, now all checks have passed :)

**Comment by guschmue** [comment]
super, thanks!

**Review by guschmue — APPROVED** [review]

**Comment by guschmue** [comment]
I was just going to merge it and noticed that we have the load_op_libraries options but we must have lost the code that processes it. One more change please :)

**Comment by zerollzeng** [comment]
Sorry @guschmue, quite busy these days, add a new commit to fix the missing part, check it by myself, please double check it again. Thanks a lot :D

BTW when I have time I can help add related docs about this argument.

**Comment by guschmue** [comment]
yeap, looks good now

**Pull request merged**